### ISSUE TITLE
"Signal 11 caught." After Connecting to Server

### ISSUE THREAD

**Issue by thomhuang** [thijsvanloef/palworld-server-docker#278]
## Describe the bug

Whenever I go to connect to my dedicated server, I hear the in-game sounds and get this in console, most notable the last three:

```
2024-02-02 21:57:18 palworld-server  | *****STARTING SERVER*****
2024-02-02 21:57:18 palworld-server  | ./PalServer.sh -port=8211 -queryport=27015
2024-02-02 21:57:18 palworld-server  | time="2024-02-03T05:57:18Z" level=info msg="read crontab: /home/steam/server/crontab"
2024-02-02 21:57:21 palworld-server  | Shutdown handler: initalize.
2024-02-02 21:57:21 palworld-server  | Increasing per-process limit of core file size to infinity.
2024-02-02 21:57:21 palworld-server  | Unable to determine CPU Frequency. Try defining CPU_MHZ.
2024-02-02 21:57:21 palworld-server  | 
2024-02-02 21:57:21 palworld-server  | [S_API] SteamAPI_Init(): Loaded local 'steamclient.so' OK.
2024-02-02 21:57:21 palworld-server  | Unable to determine CPU Frequency. Try defining CPU_MHZ.
2024-02-02 21:57:21 palworld-server  | CAppInfoCacheReadFromDiskThread took 7 milliseconds to initialize
2024-02-02 21:57:21 palworld-server  | Setting breakpad minidump AppID = 2394010
2024-02-02 21:57:21 palworld-server  | [S_API FAIL] Tried to access Steam interface SteamUser021 before SteamAPI_Init succeeded.
2024-02-02 21:57:21 palworld-server  | [S_API FAIL] Tried to access Steam interface SteamFriends017 before SteamAPI_Init succeeded.
2024-02-02 21:57:23 palworld-server  | [S_API FAIL] Tried to access Steam interface STEAMAPPS_INTERFACE_VERSION008 before SteamAPI_Init succeeded.
2024-02-02 21:57:25 palworld-server  | [S_API FAIL] Tried to access Steam interface SteamNetworkingUtils004 before SteamAPI_Init succeeded.
2024-02-02 22:00:11 palworld-server  | Signal 11 caught.
2024-02-02 22:00:11 palworld-server  | Malloc Size=262146 LargeMemoryPoolOffset=262162 
2024-02-02 22:01:24 palworld-server  | Segmentation fault (core dumped)
```

## To Reproduce

starting my container

## Expected behavior

I can connect to the server successfully

## Screenshots

also running into this issue:

<img width="865" alt="image" src="https://github.com/thijsvanloef/palworld-server-docker/assets/81591042/47fe2a3f-b643-4ab4-a1dd-415b8c79331f">

## docker-compose.yml contents

```yaml
services:
   palworld:
      image: thijsvanloef/palworld-server-docker:latest
      restart: unless-stopped
      container_name: palworld-server
      stop_grace_period: 30s # Set to however long you are willing to wait for the container to gracefully stop
      ports:
        - 8211:8211/udp
        - 27015:27015/udp # Required if you want your server to show up in the community servers tab 
      env_file:
         - .env 
      volumes:
         - ./palworld:/palworld/
      platform: linux/amd64
```

**Comment by thijsvanloef** [comment]
@thomhuang could you share some information about the hardware you are running on?

**Comment by thomhuang** [comment]
Yeah, running on MacOS Ventura, M1 Silicon, 16GB RAM, allowing 16GB ram and 8 vcpu for Docker resources. Let me know if you need any more info

**Comment by ixonstater** [comment]
I am getting the same error and am on almost exactly the same hardware as @thomhuang.

**Comment by watterso** [comment]
I ran into the same issue when developing my own docker image for Palworld, so I tested with this image and got the same issue. I was developing on an M1 mac as well and I think it must be something to do with that platform because I took the exact same command from your Readme and ran it on an AWS instance and the server ran with no issues.

TL;DR I think its an M1 architecture or a Docker on MacOS issue

**Comment by thijsvanloef** [comment]
@thomhuang From the other comments, I believe this is an issue with the M1 macbook. Sadly I cannot recreate or test this as I do not have one. I've marked it "help wanted" in the hopes someone comes up with a solution. I'm sorry I'm not able to help :(

**Comment by winstxnhdw** [comment]
I had this issue. This happens because we're using the wrong image. 

```dockerfile
services:
  palworld:
    image: thijsvanloef/palworld-server-docker:latest-arm64
```

This is the correct image. The comment that tells you to change this is so tiny that we all missed it.

**Comment by thijsvanloef** [comment]
@thomhuang Could you try to run it with this image?
```yaml
image: thijsvanloef/palworld-server-docker:latest-arm64
```

**Comment by ixonstater** [comment]
I just tried the `image: thijsvanloef/palworld-server-docker:latest-arm64` and it solved the signal 11 error, I'm able to connect to the server successfully now.
